src/ethereum/prague/utils/message.py

```diff
@@ -76,18 +79,40 @@ def prepare_message(
         Items containing contract creation or message call specific data.
     """
     if isinstance(target, Bytes0):
-        current_target = compute_contract_address(
+        current_target = compute_contract_address_1(
             caller,
             get_account(env.state, caller).nonce - U256(1),
         )
-        msg_data = Bytes(b"")
-        code = data
+        if get_eof_version(data) == EofVersion.LEGACY:
+            msg_data = Bytes(b"")
+            code = data
+            eof = None
+        else:
+            eof, msg_data = parse_create_tx_call_data(data)
+            code = eof.container
     elif isinstance(target, Address):
         current_target = target
         msg_data = data
         code = get_account(env.state, target).code
         if code_address is None:
             code_address = target
+
+        if get_eof_version(code) == EofVersion.LEGACY:
+            eof = None
+        else:
```

Comment:
Using a catch-all here will make this more difficult to catch if we add another `EofVersion` variant.

Is mypy smart enough to let us do something like:

```python
eof_version = get_eof_version(code)
if eof_version == EofVersion.LEGACY:
    ...
elif eof_version == EofVersion.EOF1:
    ...
else:
    assert_never(eof_version)
```

IIRC, `assert_never` is available in typing_extensions.